### DISCUSSION TITLE
Diagrams (-v0.18)

### DISCUSSION THREAD

**Discussion by dai-shi** [wakujs/waku#151] [General]
### Waku architecture

https://excalidraw.com/#json=ktQqatDTmpjPEkKeSqjS3,IN8mMc9-1gqNHZmHpU5sGA

![waku-arch](https://github.com/dai-shi/waku/assets/490574/ad94bbb8-7079-4cfd-a83a-aa57a5a033b1)

### How React Server Functions Work

https://excalidraw.com/#json=siXvL4HdlxlUozWFYKJPW,_Fu2i0hEjxLvzMAts07OyA

![waku-rsf](https://github.com/dai-shi/waku/assets/490574/4e633a08-0329-4f84-b18d-1f04388dc5a8)

### How Waku counter example communicates with server

https://excalidraw.com/#json=LMrRnVfDm8TDGtP-BfHZ5,o1fI7c_HvL81TDKSRoEc5A

![waku-counter](https://github.com/dai-shi/waku/assets/490574/ca5685c6-a5b2-434a-89bd-272c0d87e935)

### Waku's minimal spec for React Server Components

https://excalidraw.com/#json=sTFQvQWkoMJdkFYFJ3kc6,0HRv2uDjRNfDVOxaFIHoXg

![waku-spec](https://github.com/dai-shi/waku/assets/490574/29674fb1-a2bf-4b0d-a987-e80addaffe6f)

### How Waku's RSC-only SSR works

https://excalidraw.com/#json=pSsBQOqkYX4O-TIgxNrZj,i_CsymP1VZiHIAa0dlLbNw

![waku-ssr](https://github.com/dai-shi/waku/assets/490574/84629e46-518a-4ab1-946a-8a31c80db879)

### Waku v0.15.0 Protocol

https://excalidraw.com/#json=8muYAv1EfRMXi_If8h1Qn,ZDIumwmVrekMQqCTHLNJyg

![waku-protocol](https://github.com/dai-shi/waku/assets/490574/6fac50e3-7890-447b-ac90-8aa7d6f1ac1b)

### Waku v0.18.0 Tooling

https://excalidraw.com/#json=TSnDJo0loyNnzdbBYvlAh,EwPY-HpHy7DQmQJDKjxoPw

![waku-tool](https://github.com/dai-shi/waku/assets/490574/4cc33f66-9591-4ecf-a481-d96db38378be)

**Comment by abiriadev** [comment]
Is 'RSC Server' a separate process? (or worker?)

**Reply by dai-shi** [reply]
As of now (v0.20.2), it's a worker.

**Reply by abiriadev** [reply]
Do you have any reason to use a worker as a RSC server (e.g. `react-server` conditions)? I'm just curious!

**Reply by dai-shi** [reply]
Yes, for `react-server` conditions. We will explore to avoid the worker soon. Stay tuned.
